torch/distributed/distributed_c10d.py

```diff
@@ -234,6 +239,10 @@ def register_backend(cls, name, func, extended_api=False, devices: Optional[Unio
 
         setattr(Backend, name.upper(), name.lower())
         Backend.backend_list.append(name.lower())
+        if devices is not None:
+            for device in devices:
+                if device != 'cpu' and device != 'cuda':
```

Comment:
question (can be addressed in the next PR): why we skip for cpu and cuda, if user register a custom backend for cpu/cuda device, it is highly likely user want to use this backend for cpu/cuda, I think a default backend should also be switch to the custom registered backend?

Reply:
There are two reasons: 1.Not change the current default process when doing init_process_group without specifying backend. 2.If without the device-check, when the function 'register_backend' is called, some backends like 'fake', 'threaded' for cpu will also be added in the default_device_backend_map.
Maybe in the next pr, I can extract this fuction to a new interface. When users want to register a new backend to the default_device_backend_map,  they can call the new interface, which can perfectly solve the two problems above.